modules/ROOT/pages/_partials/commons/common-new-logging-api.adoc

```diff
@@ -0,0 +1,634 @@
+
+----
+
+.New API
+:!example-caption:
+[source, java, indent=0]
+----
+
+include::java:example$snippets/common/main/java/com/couchbase/codesnippets/Examples.java[tags=new-file-logging]
+
+----
+
+endif::is-java[]
+ifdef::is-csharp[]
+
+.Old API
+:!example-caption:
+[source, csharp, indent=0]
+----
+include::csharp:example$code_snippets/Program.cs[tags=file-logging]
+----
+
+.New API
+:!example-caption:
+[source, csharp, indent=0]
+----
+include::csharp:example$code_snippets/Program.cs[tags=new-file-logging]
+----
+endif::is-csharp[]
+ifdef::is-objc[]
+
+.Old API
+:!example-caption:
+[source, objc, indent=0]
+----
+
+include::objc:example$code_snippets/SampleCodeTest.m[tags=file-logging]
+
+----
+
+.New API
+:!example-caption:
+[source, objc, indent=0]
+----
+
+include::objc:example$code_snippets/SampleCodeTest.m[tags=new-file-logging]
+
+----
+endif::is-objc[]
+ifdef::is-swift[]
+
+.Old API
+:!example-caption:
+[source, swift]
+----
+
+include::swift:example$code_snippets/SampleCodeTest.swift[tags=file-logging]
+
+----
+
+.New API
+:!example-caption:
+[source, swift]
+----
+
+include::swift:example$code_snippets/SampleCodeTest.swift[tags=new-file-logging]
+
+----
+endif::is-swift[]
+ifdef::is-c[]
+
+.Old API
+:!example-caption:
+[source, c, indent=0]
+----
+
+include::c:example$code_snippets/main.cpp[tags=file-logging]
+
+----
+
+.New API
+:!example-caption:
+[source, c, indent=0]
+----
+
+include::c:example$code_snippets/main.cpp[tags=new-file-logging]
+
+----
+endif::is-c[]
+
+[#lbl-custom-logsink]
+=== Using a Custom Logger
+
+Installing a custom log sink with the new API is also streamlined: create an
+instance of your custom sink and use `LogSinks.get().setCustom` to install it.
+
+As with the other log sinks, you will have to specify the level and domain
+at which Couchbase logs are forwarded to your custom sink at its creation.
+
+Your custom log sink code will have to change as well.  The most significant
+change is that instead of _implementing_ the `Logger` interface, a custom log sink
+must _extend_ the `BaseLogSink` class.  
+
+The `BaseLogSink` class does not have a no-args constructor: you must specify at least the level at which Couchbase logs will be forwarded to the logger.  
+
+A second important change is that your logger will receive only logs at the level and domain for which it is initialized. 
+There is no need to record or filter the logs forwarded to the protected `writeLog` method which replaces the public `log` method from the old API.
+
+Related to this last point, the Couchbase `Loggers`, now `LogSinks` are meant to support logging by the Couchbase Lite platform.
+They were never meant as a general framework for logging. 
+
+IMPORTANT: With the new API, customer code can no longer log, directly, to any of the Couchbase log sinks.  
+The Console and File logsinks are final classes and do not publish methods that allow writing logs.
```

Comment:
Although, from the user perspective, both a final class and a struct cant be subclassed, this is not true for iOS. Swift LogSinks are structs and Obj-C doesn't have a final class concept (we "manually" prevent against subclassing.)

Reply:
Reworded "cannot be subclassed"